docs/css/README.md

```diff
@@ -4,36 +4,134 @@
+
+### Take advantage of the cascade
+
+<Versus>Don’t set identical, inheritable properties like <code>color</code>, <code>font-…</code>, <code>direction</code>, … on many sibling elements</Versus>
+<Versus correct>Set inheritable properties targeting multiple children on a common parent</Versus>
+
+### Enjoy utility classes
+
+<Versus>Avoid cluttering your CSS with many variants of the same pattern: <code>.list-inline--right</code>, <code>.media--right</code>, <code>.actions--right</code></Versus>
+<Versus correct>Use a single utility classes to alter the behavior: <code>.text-right</code>, <code>.items-end</code>, <code>.text-gray-500</code></Versus>
+
+### Let the parent handle outer spacing
+
+Does this element should really always have a `margin-bottom`? Or is it only in some specific context?
+
+A rule of thumb you can apply in 99% of cases is: **a component should only care about its own inner spacing**.
+
+* Need space between items in a list? Use `gap` on the parent.
+* Need space around columns and rows in a grid? Use `gap` on the parent.
+* Need space between a title and a paragraph? Add `margin-bottom` on the title or `margin-top` on the paragraph depending if one or the other is optional, but do it **only** for this title or this paragraph in particular (using utility classes for example). Alternatively, add a wrapper around both and use… `gap`!
+
+> It’s gaptastic!
+>
+> — [Eric Meyer](https://meyerweb.com/)
+
+### Augment rather than deconstruct
+
+<Versus>Undoing existing style can be tedious, reduce composability and requires more code:</Versus>
+
+```css
+.card {
+  display: flex;
+  background-color: var(--light-blue);
+}
+
+.card--no-background {
+  background-color: transparent;
+}
+```
+
+<Versus correct>Create additionnal variants and compose them instead:</Versus>
+
+```css
+.card {
+  display: flex;
+}
+
+.card--default {
+  background-color: var(--light-blue);
+}
+```
+
+<Versus>Redefining the same value multiple times at different places can leads to unexpected behaviors and bugs:</Versus>
+
+```css
+.btn {
+  display: flex;
+}
+```
+
+In the HTML below, the `sm:flex` style duplicates the value from the CSS above:
+
+```html
+<button class="btn hidden sm:flex"></button>
+```
+
+<Versus correct>Avoid duplicates by altering only the necessary properties:</Versus>
+
+```html
+<button class="btn max-xs:hidden"></button>
+```
+
+### Use a mobile-first approach
+
+When they do not imply deconstructing or duplicating code as stated above, you should always use `min-width` media queries first and alter style from the smallest screen and up.
+
+### Use shorthands responsibly
+
+Please read [CSS Shorthand Syntax Considered an Anti-Pattern](https://csswizardry.com/2016/12/css-shorthand-syntax-considered-an-anti-pattern/) for a detailed explanation.
+
+### Group and order declarations
```

Comment:
Should we add nesting to that section (or maybe another one) ? I think there is a lot going on with things such as `modifiers`, `pseudo-elements`, `pseudo-classes` or `@media` declarations... I clearly see some benefit to be consistent here.

Reply:
A second point about nesting is how we handle `modifiers`. There are multiple options to handle them in our declarations and I would find it great to define a standardized way applicable most of the time. That's particularly true with the `BEM` syntax.

```pcss
/*
 * Modifiers are considered as first-class citizen in our CSS declarations,
 * we emphasis the state of the application in our style definitions.
 */

.navbar {
  background-color: white;
}

.navbar__link {
  color: black;
}

.navbar--expanded {
  background-color: black;

  .navbar__link {
    color: black;
  }
}

/*
 * Modifiers are nested under blocs or elements declarations,
 * we emphasis the component encapsulation of the style definitions
 */

.navbar {
  background-color: white;
 
  &.navbar--expanded {
    background-color: black;
  }
}

.navbar__link {
  color: black;

  .navbar--expanded & {
    color: black;
  }
}
```

Reply:
The aim is to reduce specificity as much as we can so that the codebase remains maintainable/scalable as much as possible. So, from my point of view, nesting should only be used if increasing specificity is required.